src/components/axes/timeAxis.ts

```diff
@@ -207,6 +202,19 @@ export module Axis {
         return this._possibleTimeAxisConfigurations;
       }
       this._possibleTimeAxisConfigurations = configurations;
+      this._numTiers = _Util.Methods.max(this._possibleTimeAxisConfigurations.map((config: TimeAxisConfiguration) => config.length), 0);
+
+      if (this._isAnchored) {
+        this._createDomElements();
+      }
+
+      var oldLabelPositions: string[] = this.tierLabelPositions();
+      var newLabelPositions: string[] = [];
+      for (var i = 0; i < this._numTiers; i++) {
+        newLabelPositions.push(oldLabelPositions[i] || "between");
+      }
+      this.tierLabelPositions(newLabelPositions);
```

Comment:
I would probably expect the latter. But then every time we use the tierLabelPositions we have to 1) Check for the size of the array 2) Check for the attribute, eventually resize the array. Also, the number of tiers is always 1 or 2, this method is not called that often, so I think it is acceptable to recreate an array of 2 elements all the time